.github/workflows/test-servers.yml

```
@@ -0,0 +1,74 @@
+    branches: [main]
+  workflow_call:
+
+jobs:
+  detect-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      assessment-changed: ${{ steps.changes.outputs.assessment }}
+      core-changed: ${{ steps.changes.outputs.core }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v2
```

actionlint (1.7.7)

the runner of "dorny/paths-filter@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)